app/src/lib/git/cherry-pick.ts

```diff
@@ -187,11 +194,23 @@ function parseCherryPickResult(result: IGitResult): CherryPickResult {
       return CherryPickResult.ConflictsEncountered
     case GitError.UnresolvedConflicts:
       return CherryPickResult.OutstandingFilesNotStaged
+    case GitError.ConflictModifyDeletedInBranch:
+      throw new Error(parseConflictModifyDeleteFilesError(result))
     default:
       throw new Error(`Unhandled result found: '${JSON.stringify(result)}'`)
   }
 }
 
+function parseConflictModifyDeleteFilesError(result: IGitResult): string {
+  const files = parseConflictModifyDeleteFiles(result.stdout)
+  const commitSummary = parseConflictModifyDeleteCommitSummary(result.stdout)
+  let error = `Cherry-pick failed. The commit with summary '${commitSummary}' modifies one or more files that does not exist on the target branch.`
+  files.forEach(file => {
+    error = error + '\n' + file
+  })
+  return error
```

Comment:
If we assume at this point `files` won't be empty (that's the whole reason for this PR, right? 🤔 ) I guess we can simplify the logic like this:

```suggestion
  let error = `Cherry-pick failed. The commit with summary '${commitSummary}' modifies one or more files that does not exist on the target branch.`
  return error + '\n' + files.join('\n')
```